### PR TITLE
feat(orc8r): Changes to accept AMF configs from orc8r

### PR DESCRIPTION
**Pull request by shivesh-wavelabs**
## Summary

This PR deals with the changes as mentioned in the [document ](https://docs.google.com/document/d/1yywmytLNr8p704m7rFcAfHGanG90ZeIo/edit#)

Added updates to swagger.v1.yml to accept AMF configs.
Modified Mconfig builder to fill the mconfig message with New attributes
Added/Modified obsidian handlers to accept AMF configs.

AGW and proto changes are in [#10392](https://github.com/magma/magma/pull/10392).

### Test Plan

Added UT cases in obsidian handlers to verify the new endpoints and new attributes for added AMF attributes.

### CODE REVIEW

File: lte/cloud/go/services/lte/obsidian/handlers/handlers_test.go [hunk 1]
```diff
@@ -1199,13 +1199,15 @@ func TestListAndGetGateways(t *testing.T) {
 				Config: &lteModels.GatewayCellularConfigs{
 					Epc: &lteModels.GatewayEpcConfigs{NatEnabled: swag.Bool(true), IPBlock: "192.168.0.0/24"},
 					Ran: &lteModels.GatewayRanConfigs{Pci: 260, TransmitEnabled: swag.Bool(true)},
+					Ngc: &lteModels.GatewayNgcConfigs{AmfDefaultSd: "AFAFAF", AmfDefaultSst: swag.Uint32(25), AmfName: "amf.example.org", AmfPointer: "1F", AmfRegionID: "C1", AmfSetID: "2A1"},
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/obsidian/handlers/handlers_test.go]: Nit: here and elsewhere: let's break these long-line struct literals into multiple lines -- one field per line

File: lte/cloud/go/services/lte/servicers/builder_servicer.go [hunk 2]
```diff
@@ -184,6 +185,14 @@ func (s *builderServicer) Build(ctx context.Context, request *builder_protos.Bui
 			SentryConfig:             getNetworkSentryConfig(&network),
 			Enable5GFeatures:         swag.BoolValue(nwEpc.Enable5gFeatures),
 		},
+		"amf": &lte_mconfig.AMF{
+			AmfName:                    gwNgc.AmfName,
+			AmfSetId:                   gwNgc.AmfSetID,
+			AmfRegionId:                gwNgc.AmfRegionID,
+			AmfPointer:                 gwNgc.AmfPointer,
+			DefaultSliceServiceType:    *gwNgc.AmfDefaultSst,
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/servicers/builder_servicer.go]: This feels dangerous. Either make the field non-nil (in swagger), or guard against the field potentially being nil

**Review by hcgatewood — CHANGES_REQUESTED** [review]
Overall lgtm. 2 small things then I'll be content

File: lte/cloud/go/services/lte/servicers/builder_servicer.go (hunk 2 above)
Comment (shivesh-wavelabs) [reply lte/cloud/go/services/lte/servicers/builder_servicer.go]: Sure, I'll make `x-nullable:false` for these in swagger.

File: lte/cloud/go/services/lte/obsidian/handlers/handlers_test.go (hunk 1 above)
Comment (shivesh-wavelabs) [reply lte/cloud/go/services/lte/obsidian/handlers/handlers_test.go]: Sure.

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 3]
```diff
@@ -1066,6 +1068,40 @@ paths:
           description: Success
         default:
           $ref: './orc8r-swagger-common.yml#/responses/UnexpectedError'
+
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: Fix yamllint errors

File: lte/cloud/go/services/lte/servicers/builder_servicer.go (hunk 2 above)
Comment (hcgatewood) [reply lte/cloud/go/services/lte/servicers/builder_servicer.go]: Looks like this is still a ptr that we're not nil-guarding (?)

**Review by hcgatewood — COMMENTED** [review]
2 comments, not sure on the latter one so feel free to DM me if I'm misunderstanding. Otherwise lgtm -- lmk and I'll approve!

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 3 above)
Comment (shivesh-wavelabs) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: okay.

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 4]
```diff
@@ -2176,6 +2214,41 @@ definitions:
...
+        type: string
+        pattern: '^[0-3][A-Fa-f0-9]{1}$'
+        x-nullable: false
+        description: String identifying the AMF Pointer (6 bits) as specified in clause 2.10.1 of 3GPP TS 23.003.
+      amf_default_sst:
+        type: integer
+        format: uint32
+        minimum: 0
+        maximum: 255
+        description: AMF Default Slice Service Type. Values 0 to 127 correspond to the standardized SST range. Values 128 to 255 correspond to the Operator-specific range. See clause 28.4.2 of 3GPP TS 23.003. Standardized values are defined in clause 5.15.2.2 of 3GPP TS 23.501.
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: @shivesh-wavelabs this is the particular object I believe you need to mark as x-nullable false

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 5]
```diff
@@ -2176,6 +2214,41 @@ definitions:
...
+        pattern: '^[A-Fa-f0-9]{1,2}$'
+        x-nullable: false
+        description: String identifying the AMF Region ID (8 bits) as specified in clause 2.10.1 of 3GPP TS 23.003.
+      amf_pointer:
+        type: string
+        pattern: '^[0-3][A-Fa-f0-9]{1}$'
+        x-nullable: false
+        description: String identifying the AMF Pointer (6 bits) as specified in clause 2.10.1 of 3GPP TS 23.003.
+      amf_default_sst:
+        type: integer
```
Comment (hcgatewood) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2238]:
```suggestion
      amf_default_sst:
        type: integer
        x-nullable: false
```

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 4 above)
Comment (shivesh-wavelabs) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml]: got it.

**Review by hcgatewood — APPROVED** [review]
Lgtm 👍

**Comment by anusha-konda** [comment]
Looks like each section in gateway.mconfig represents a service ( like mme, pipelined) in AGW, whereas AMF is launched as thread.
Introducing a separate section for amf is not inline with the current framework to read configs from gateway.mconfig by a service.
So will add all the new attributes in mme section itself and make it configurable from swagger.
cc @panyogesh, @ulaskozat

**Comment by shivesh-wavelabs** [comment]
Hi @hcgatewood, can you re-review this PR ?

**Review by hcgatewood — APPROVED** [review]
Lgtm! 💯💯

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
proto and pb files are in #10222, right? Please update the summary about the dependence.

**Comment by shivesh-wavelabs** [comment]
@hcgatewood could you merge this PR, its dependent PR #10392 is already merged.

**Comment by hcgatewood** [comment]
@shivesh-wavelabs looks like the insync job is hanging. Can you rebase onto latest master and retry, should fix

**Comment by shivesh-wavelabs** [comment]
> @shivesh-wavelabs looks like the insync job is hanging. Can you rebase onto latest master and retry, should fix

done.

**Pull request merged**